src/inventorymanager.h

```diff
@@ -183,6 +183,14 @@ struct IMoveAction : public InventoryAction, public MoveAction
 	void apply(InventoryManager *mgr, ServerActiveObject *player, IGameDef *gamedef);
 
 	void clientApply(InventoryManager *mgr, IGameDef *gamedef);
+
+	void swapDirections();
+
+	void onPutAndOnTake(ItemStack &src_item, ServerActiveObject* player);
```

Comment:
Please check if (for all three methods):
* the `ItemStack` argument can be const
* the method can be marked const

Reply:
Please also correct the pointer placement: `ServerActiveObject *player`.